### PR TITLE
[timeseries] Add scale feature and normalize covariates for MLForecast models

### PR DESCRIPTION
**Pull request by shchur**
*Description of changes:*
- Some lightweight feature engineering for MLForecast-based models:
  - Add two static features, encoding mean & std of the target time series for each item
  - For each continuous real-valued covariate, add a copy of this covariate that is mean-abs-scaled per item

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py [hunk 1]
```diff
@@ -252,6 +266,11 @@ def _to_mlforecast_df(
         if static_features is not None:
             df = pd.merge(df, static_features, how="left", on=ITEMID, suffixes=(None, "_static_feat"))
 
+        for col in self.metadata.known_covariates_real:
+            # Normalize non-boolean features using mean_abs scaling
+            if not df[col].isin([0, 1]).all():
+                df[f"__scaled_{col}"] = df[col] / df[col].abs().groupby(df[ITEMID]).mean().reindex(df[ITEMID]).values
```
Comment (canerturkmen) [inline timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py:273]: nice pandas sorcery :)

**Review by canerturkmen — APPROVED** [review]
LGTM!

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Pull request merged**